src/main/java/ro/societateahermes/backendservice/service/serviceImplementation/UserServiceImplementation.java

```diff
@@ -22,6 +23,13 @@ public void save(User user) {
         userRepository.save(user);
     }
 
+    @Override
+    public User saveUserFromDTO(MySubmissionDTO submissionDTO) {
+
+        User user =new User(submissionDTO.getFirstName(),submissionDTO.getLastName(),submissionDTO.getEmail(),submissionDTO.getUsername(),submissionDTO.getPassword(),submissionDTO.getUniversity());
+        return userRepository.save(user);
+    }
+
     @Override
     public List<User> getAllUsers() {
         return userRepository.findAll();
```

Comment:
dto. Overall codul este de foarte buna calitate, dar toate remarcile pe care ti le-am facut aici trebuiesc rezolvate

Reply:
Am mapat in dto in controller, si metodele din service le-am lasat fara dto.....in cazul asta ramane List<User> get AllUsers() dar in UserController returnez lista asta mapata la List<UserDTO>

Cred ca am facut si restul modificarilor cerute. Cand ai timp sa te uiti daca e totul in regula si pot da merge.

Reply:
Sigur ti-ai dat seama pana acum :))) dar maperea nu are ce sa caute in controller si ar trebui sa ai o clasa separata care se ocupa cu asa ceva. Daca faci asta atunci controllerul nu o sa mai aibe o singura responsabilitate, iar asta va face viata mai grea mai tarziu in ciclul de viata al aplicatiei

Reply:
Maparea am facut-o dupa https://www.javaguides.net/2021/02/spring-boot-dto-example-entity-to-dto.html, folosind modelMapper si asa nu scriu eu cate o metoda pentru fiecare convertire si o face automat, prin determinarea automată a modului în care un model de obiect mapează la altul, pe baza convențiilor........ astfel am adugat dependentele necesare pentru el in pom, bean ul necesar in main 
    @Bean
    public ModelMapper modelMapper() {
        return new ModelMapper();
    }
si in controller am injectat ModelMaper prin
@Autowired
 private ModelMapper modelMapper
Si am adauagt partea asta in controller: map(user -> modelMapper.map(user, UserDTO.class)) atunci cand vreau sa mapez user la userDTO de exemplu
Dar o sa incerc sa mut toate astea in service, mi se pare o idee buna sa gestionez astfel maparile in loc sa fac o clasa de mapare pentru fiecare entitate. Daca e mai ok a doua, cea cu clasele o sa incerc sa le fac pe toate asa, am vrut doar sa explic cum am gandit si cum am gestionat toata partea asta ( si de ce am facut-o asa)